Only Admin Profile Users can add/edit/delete other profiles

kriscfoster opened the pull request:
W-4175897
Will now require an admin access token to create/edit/delete a profile. Also added some tests for this.

kriscfoster commented:
@iamigo I can now confirm that all of those scenarios are tested for. I added a new routine called "test-api-profiles" which is invoked with REQUIRE_ACCESS_TOKEN=true.. Let me know whether this was the correct way to go about this.

iamigo commented:
@kriscfoster not another new one... pls. use the "test-enforced" script in package.json

kriscfoster commented:
@iamigo Didn't see that, thanks.. I have made those changes now

iamigo commented on package.json:
```diff
@@ -41,7 +41,7 @@
     "test-api-log": "ENABLE_API_ACTIVITY_LOGS=true mocha -R dot --recursive tests/api",
     "test-disablehttp": "REQUIRE_HTTPS=true mocha -R dot --recursive tests/disableHttp",
-    "test-enforced": "REQUIRE_ACCESS_TOKEN=true mocha -R dot --recursive tests/enforceToken",
+    "test-enforced": "REQUIRE_ACCESS_TOKEN=true mocha -R dot --recursive tests/enforceToken tests/api/v1/profiles",
```
So adding this here, we're now executing all the tests under tests/api/v1/profiles _once_ with REQUIRE_ACCESS_TOKEN=true and again with REQUIRE_ACCESS_TOKEN=false? @kriscfoster

kriscfoster replied:
That's correct, all tests under v1/profiles are being called once with REQUIRE_ACCESS_TOKEN = true and once with false @iamigo

iamigo commented:
I'm ready to approve this once tests are passing.

shriramshankar commented on tests/api/v1/profiles/post.js:
```diff
@@ -45,11 +55,36 @@ describe(`api: POST ${path}`, () => {
+      .expect(constants.httpStatus.FORBIDDEN)
+      .end((err, res) => {
+        if (err) {
+          done(err);
```
You don't have to change it in this PR but we ideally should be adding a "return" in front of done(err). If you don't add it, all the errors will fail, saying that "done" was called twice.

kriscfoster commented:
Added returns before done(err)

In tests/api/v1/profiles/get I changed the expectations for admins to have 'rw' access for imc resources because this may cause confections with yesterdays migration which initialized all of the new IMC resource access's to 'r' as default.
